ObjectMapper/Core/Mapper.swift

```diff
@@ -102,6 +102,13 @@ public final class Mapper<N: BaseMappable> {
 			}
 		}
 
+		if let klass = N.self as? ImmutableMappable.Type {
+			if let maybeObject = try? klass.init(map: map) as? N, var object = maybeObject {
```

Comment:
I wanted to replicate the behaviour of the `StaticMappable` handling as closely as possible - if the initializer fails for some reason, execution will continue on to the "fallback" if statement below. Alternatively I could just return `nil`, which seems reasonable because I think `ImmutableMappable` and `Mappable` are mutually exclusive - you wouldn't have an object that conforms to both (and from my quick experiments, such a setup won't compile).